app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt

```diff
@@ -44,4 +54,6 @@ object StartupTimeline {
     private fun advanceState(startingActivity: StartupActivity) {
         state = StartupTimelineStateMachine.getNextState(state, startingActivity)
     }
+
+    fun <R> measureGeckoRuntimeCreate(block: () -> R): R = measurements.geckoRuntimeCreate.measure { block() }
```

Comment:
Not really a problem, but more of a curiosity: is there any advantage in introducing this function over calling something like `measure(gleanMetric, block)`? It would save you the trouble of defining a function for each segment you want to measure.

Reply:
I love this solution! Great idea!